bots/pr/src/main/java/org/openjdk/skara/bots/pr/PullRequestCommandWorkItem.java

```diff
@@ -40,9 +39,8 @@ public class PullRequestCommandWorkItem extends PullRequestWorkItem {
 
     private static final String commandReplyMarker = "<!-- Jmerge command reply message (%s) -->";
     private static final Pattern commandReplyPattern = Pattern.compile("<!-- Jmerge command reply message \\((\\S+)\\) -->");
-    private final static Pattern pushedPattern = Pattern.compile("Pushed as commit ([a-f0-9]{40})\\.");
 
-    private static final Map<String, CommandHandler> commandHandlers = Map.ofEntries(
+    static final Map<String, CommandHandler> commandHandlers = Map.ofEntries(
```

Comment:
Instead of making these package private and adding the convoluted logic in extractCommands, can't we just add BackportCommand to the commandHandlers for PullRequestWorkItem?

It seems like the original author mixed two ways of separating commands between PRs and commits, both through these handler lists, and through the allowedIn* methods. If we really want to improve all the error messages, then all commands should be added to both lists, or maybe the lists should be unified into one.

Reply:
I unified these two lists into one and put it in class `CommandExtractor`. All the tests in PR bot passed locally.